python/lsst/ts/envsensors/socket_server.py

```diff
@@ -55,109 +57,106 @@ def __init__(self, port, simulation_mode=0):
 
         self.simulation_mode = simulation_mode
 
-        self._server = None
-        self._writer = None
-        self._reader = None
-        self._started = False
-        self.log = logging.getLogger(type(self).__name__)
+        self._server: tcpip.OneClientServer = None
```

Comment:
I suggest subclassing OneClientServer instead of containing one, because:
* This acts like a server.
* Your class has a few very focused and relevant additions to OneClientServer: the read loop and write method, plus the connection callback to manage them (which could be be a private method, though you cannot call it `_connect_callback` because that will collide with the `_connection_callback` instance variable in OneClientServer).
* You are copying a reasonable amount of the API into this class, for example the disconnect and exit methods just call a single method each in OneClientServer, and it is a confusingly different name. Also `self.port = self._server.port`.
* You can get rid of the `start` method.

If you prefer to contain the server then I suggest:
* Construct self._server in the constructor, not in the start method. This eliminates any need to test for `self._server` being None.
* Eliminate the `start` method.
* Make `start_task` and `port` accessors on the OneClientServer.

Reply:
Subclassing OneClientServer is what I should have done from the start. Done.